dist/tools/kconfiglib/Makefile

```diff
@@ -4,11 +4,9 @@ PKG_VERSION=82f59179b1b35fcd8b6d188453b283599ea70518
 PKG_LICENSE=ISC
 PKG_BUILDDIR=$(CURDIR)/bin
 
-.PHONY: all
+include $(RIOTBASE)/pkg/pkg.mk
 
-all: git-download
+all:
```

Comment:
I think that now since the include `include $(RIOTBASE)/pkg/pkg.mk` before `all` none of the `tools.` need to depend on `git-download`, although you commented:

Nevermind, it seems that it is needed if you properly clean the repo ;-)

_Originally posted by @basilfx in https://github.com/RIOT-OS/RIOT/pull/13032#issuecomment-570961309_

How did you test this? I want to try and replicate for other pkgs.

Reply:
Initial tests seem to confirm it is not needed.

Reply:
Please make sure that the dist/tools/kconfiglib is clean (`git clean -fxd`).

Reply:
Also works on my machine:

```
basilfx:pkg_utensor/ (feature/save_state_and_dependencies) $ QUIET=0 BOARD=slstk3401a make                                                                                                         [INFO] Kconfiglib not found - getting it                                                                                                                                                           make[1]: Entering directory '/home/basilfx/RIOT/RIOT/dist/tools/kconfiglib'                                                                                                                        git -C /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin --git-dir=.git --work-tree=. fetch  https://github.com/ulfalizer/Kconfiglib 82f59179b1b35fcd8b6d188453b283599ea70518                      From https://github.com/ulfalizer/Kconfiglib                                                                                                                                                        * branch                82f59179b1b35fcd8b6d188453b283599ea70518 -> FETCH_HEAD                                                                                                                    echo 82f59179b1b35fcd8b6d188453b283599ea70518 > /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin/.pkg-state.git-downloaded                                                                        git -C /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin --git-dir=.git --work-tree=. clean  -xdff '**' ':!.pkg-state.git*'                                                                        Removing .git-downloaded                                                                                                                                                                           git -C /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin --git-dir=.git --work-tree=. checkout  -f 82f59179b1b35fcd8b6d188453b283599ea70518                                                        HEAD is now at 82f59179b Make 13.3.0 release with improved header comment customization                                                                                                            git -C /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin --git-dir=.git --work-tree=. -c user.email=buildsystem@riot -c user.name="RIOT buildsystem" am  --no-gpg-sign --ignore-whitespace --whitespace=nowarn  </dev/null                                                                                                                                                                            cp /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin/kconfiglib.py /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin/menuconfig.py \                                                                  /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin/genconfig.py /home/basilfx/RIOT/RIOT/dist/tools/kconfiglib/bin/examples/merge_config.py \                                                        .                                                                                                                                                                                               make[1]: Leaving directory '/home/basilfx/RIOT/RIOT/dist/tools/kconfiglib'                                                                                                                         [INFO] Kconfiglib downloaded
```

Reply:
```
  ~/workspace/RIOT3/dist/tools/kconfiglib  |pr-13036 ✓|
→ git clean -xdf
Removing __pycache__/
Removing genconfig.py
Removing kconfiglib.py
Removing menuconfig.py
Removing merge_config.py
```

```
make -C tests/kconfig/
make: Entering directory '/home/francisco/workspace/RIOT3/tests/kconfig'
[INFO] Kconfiglib not found - getting it
cp /home/francisco/workspace/RIOT3/dist/tools/kconfiglib/bin/kconfiglib.py /home/francisco/workspace/RIOT3/dist/tools/kconfiglib/bin/menuconfig.py \
   /home/francisco/workspace/RIOT3/dist/tools/kconfiglib/bin/genconfig.py /home/francisco/workspace/RIOT3/dist/tools/kconfiglib/bin/examples/merge_config.py \
   .
[INFO] Kconfiglib downloaded
Merged configuration '/home/francisco/workspace/RIOT3/tests/kconfig/app.config'
Configuration saved to '/home/francisco/workspace/RIOT3/tests/kconfig/bin/native/generated/merged.config'
Merged configuration '/home/francisco/workspace/RIOT3/tests/kconfig/app.config'
No change to '/home/francisco/workspace/RIOT3/tests/kconfig/bin/native/generated/merged.config'
Building application "tests_kconfig" for "native" with MCU "native".

"make" -C /home/francisco/workspace/RIOT3/boards/native
"make" -C /home/francisco/workspace/RIOT3/boards/native/drivers
"make" -C /home/francisco/workspace/RIOT3/core
"make" -C /home/francisco/workspace/RIOT3/cpu/native
"make" -C /home/francisco/workspace/RIOT3/cpu/native/periph
"make" -C /home/francisco/workspace/RIOT3/cpu/native/stdio_native
"make" -C /home/francisco/workspace/RIOT3/cpu/native/vfs
"make" -C /home/francisco/workspace/RIOT3/drivers
"make" -C /home/francisco/workspace/RIOT3/drivers/periph_common
"make" -C /home/francisco/workspace/RIOT3/sys
"make" -C /home/francisco/workspace/RIOT3/sys/auto_init
"make" -C /home/francisco/workspace/RIOT3/sys/test_utils/interactive_sync
   text	   data	    bss	    dec	    hex	filename
  24565	    580	  47664	  72809	  11c69	/home/francisco/workspace/RIOT3/tests/kconfig/bin/native/tests_kconfig.elf
make: Leaving directory '/home/francisco/workspace/RIOT3/tests/kconfig'
```

Reply:
Looks good!

Reply:
I would add a separate commit for all `dist/tools` though.

I also tested on master and you were right, it is needed there, but with including the `pkg` earlier it works :D

Reply:
I'll make sure that a separate commit is there when I can squash and rebase all commits.

Reply:
Made separate commits for dist/tools/*.